lib/libc/syslog/lib_syslog.c

```diff
@@ -154,15 +158,30 @@ static inline int vsyslog_internal(FAR const char *fmt, va_list ap)
 
 	lib_rawoutstream(&stream, 1);
 
+#if CONFIG_NFILE_STREAMS > 0 && (defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__))
+	if(!up_interrupt_context() && getpid() != 0){
+		lib_take_semaphore(stdout);
```

Comment:
I think below implementation is better in terms of readability and understanding.
```
#if CONFIG_NFILE_STREAMS > 0 && (defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__))
	if(!up_interrupt_context() && getpid() != 0){
		lib_take_semaphore(stdout);
       }
#endif
/* All print statements here */

#if CONFIG_NFILE_STREAMS > 0 && (defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__))
	if(!up_interrupt_context() && getpid() != 0){
		lib_give_semaphore(stdout);
       }
#endif
```

Reply:
Yeah, this improves readability.